### DISCUSSION TITLE
CUDA

### DISCUSSION THREAD

**Discussion by kangpyo** [JeffFessler/MIRT.jl#105] [General]
does MIRT implement CUDA/GPU run?

**Comment by JeffFessler** [comment]
Good question - I have not tried it yet.

Certainly the goal is to support GPU use.
Towards that end, most of the methods are designed to use `AbstractArray` rather than, say, `Matrix`,
in hopes that if you have data in a CuArray then it will "just work"
https://cuda.juliagpu.org/stable/usage/array/#Construction-and-Initialization
If you try it and find any methods do not work with data on a GPU then please report an issue.

If you are asking whether any of the internal methods use GPU kernels, then the answer is not yet,
although the NUFFT operations are built on the NFFT.jl package and I think that package has GPU support.
